File: src/main/java/com/protryon/jasm/stage1/constant/CLong.java
package com.protryon.jasm.stage1.constant;

import java.io.DataOutputStream;
import java.io.IOException;

import com.protryon.jasm.stage1.CConstant;

public class CLong extends CConstant<Long> {
	public CLong(long lng) {
		super(lng, 5);
	}

	@Override
	public void write(DataOutputStream out) throws IOException {
		out.writeLong(this.value);
	}

	public boolean isDoubled() {
		return true;
	}
}

File: src/main/java/com/protryon/jasm/stage1/constant/CLong.kt
package com.protryon.jasm.stage1.constant

import java.io.DataOutputStream
import java.io.IOException

import com.protryon.jasm.stage1.CConstant

class CLong(lng: Long) : CConstant<Long>(lng, 5) {

    @Throws(IOException::class)
    public override fun write(out: DataOutputStream) {
        out.writeLong(this.value)
    }

    override val isDoubled: Boolean
        get() = true

}
